lib/PSR/Streaming.hs

```diff
@@ -145,6 +149,24 @@ traceChainSyncEvent events = \case
         events.addSelectionEvent header
     _ -> pure ()
 
+traceTransactionExecutionResult :: Events -> TransactionContext era -> IO ()
+traceTransactionExecutionResult events tc = 
+    forM_ (Map.elems tc.ctxTransactionExecutionResult) $ \case
+      Right (pwc, logs, exUnits) -> addEvent pwc logs exUnits Nothing
+      Left (C.ScriptErrorEvaluationFailed (C.DebugPlutusFailure evalErr pwc exUnits logs)) -> addEvent pwc logs exUnits (Just evalErr) 
+      _ -> pure ()
+    where
+      addEvent PlutusWithContext{..} logs exUnits evalError' = do
+        let scriptHash = C.ScriptHash pwcScriptHash
+        events.addExecutionEvent tc.ctxBlockHeader $ ExecutionEventPayload
+          { transactionHash = C.getTxId $ C.getTxBody tc.ctxTransaction
```

Comment:
I'm so used to writing code without record dot syntax, but it makes things super readable - please remind me in reviews when I should be using it to clean things up.